### PR TITLE
Issue - 115: Added the jupyter notebook cmd env variable

### PR DESCRIPTION
**Pull request by nehiljain**
Changes proposed in this pull request:
- Add an env variable to override jupyter notebook command

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

Thank you for submitting!

### CODE REVIEW

**Comment by nehiljain** [comment]
@cla-bot check

**Comment by nehiljain** [comment]
@cla-bot check

**Comment by kyleaton** [comment]
@cla-bot check

**Review by ayirplm — COMMENTED** [review]
LGTM! Thank you for contributing!

**Review by Aylr — CHANGES_REQUESTED** [review]
This is great! Where would you have looked for something like this in the docs? If there's an obvious place, would you add a sentence or two?

**Review by jcampbell — CHANGES_REQUESTED** [review]
Thank you -- super excited about this. Could you please add a note in the changelog about the change, along the lines of "* Add support for overriding the default jupyter command via a GE_JUPYTER_COMMAND environment variable."

**Review by eugmandel — CHANGES_REQUESTED** [review]
Looks good to me, but let's add a test. Thank you!

**Comment by nehiljain** [comment]
> Looks good to me, but let's add a test. Thank you!

Added tests. Please review

**Comment by nehiljain** [comment]
> Thank you -- super excited about this. Could you please add a note in the changelog about the change, along the lines of "* Add support for overriding the default jupyter command via a GE_JUPYTER_COMMAND environment variable."

Add this to the changelog. Please review

**Comment by nehiljain** [comment]
https://travis-ci.org/github/great-expectations/great_expectations/jobs/682483716#L966 The mocking patch is not working with pytest. Is there something I am missing here?

**Comment by ayirplm** [comment]
> > Thank you -- super excited about this. Could you please add a note in the changelog about the change, along the lines of "* Add support for overriding the default jupyter command via a GE_JUPYTER_COMMAND environment variable."
> 
> Add this to the changelog. Please review

thanks @nehiljain pushing this up for review , @jcampbell @eugmandel

File: docs/changelog.rst [hunk 1]
```diff
@@ -31,6 +32,9 @@ develop
 -----------------
 
 * improvements to ge.read_json tests
+<<<<<<< HEAD:docs/changelog/changelog.rst
+* added support for overriding the default jupyter command via a GE_JUPYTER_COMMAND environment variable
+=======
```
Comment (Aylr) [inline docs/changelog.rst]:
```suggestion
```
here's a tiny merge conflict that can simply be removed.

File: docs/changelog.rst [hunk 2]
```diff
@@ -43,6 +47,7 @@ develop
 * Clarified scaffolding language in scaffold notebook
 * DataContext.create() adds an additional directory: `checkpoints`
 * Marked tap command for deprecation in next major release
+>>>>>>> bd57b9f537ec597807e2824b925b01c4e166208d:docs/changelog.rst
```
Comment (Aylr) [inline docs/changelog.rst]: Here's the rest of it that can be dropped.
```suggestion
```

File: docs/getting_started/typical_workflow.rst [hunk 3]
```diff
@@ -216,6 +216,7 @@ Edit
 
 Editing an Expectation Suite means adding, removing, and modifying the arguments of existing expectations.
 
+.. note:: If you wish to override the default jupyter notebook command then you can set GE_JUPYTER_CMD environement variable. This is specially useful for running great expectations inside dockerized environements. For dockerized GE, use `export GE_JUPYTER_CMD='jupyter notebook --ip 0.0.0.0'`
```
Comment (Aylr) [inline docs/getting_started/typical_workflow.rst:219]: Nice!

File: tests/cli/test_toolkit.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    except Exception:
+        if temp:
+            os.environ['GE_JUPYTER_CMD'] = temp
+        raise
+
+@mock.patch("subprocess.call", return_value=True, side_effect=None)
+def test_launch_jupyter_notebook_env_set_in_env(mock_subprocess):
+    with mock.patch.dict(os.environ, {'GE_JUPYTER_CMD': 'jupyter notebook --test-args'}, clear=True):
+        toolkit.launch_jupyter_notebook('test_path')
+        mock_subprocess.assert_called_once_with("jupyter notebook --test-args test_path", shell=True)
```
Comment (Aylr) [inline tests/cli/test_toolkit.py:27]: Nice work. Testing these kinds of things is tricky!

**Review by Aylr — APPROVED** [review]

File: tests/cli/test_toolkit.py (hunk 4 above)
Comment (nehiljain) [reply tests/cli/test_toolkit.py:27]: I didnt like my mutation of environ dict to set it to none :( but didnt find a good solution right away

**Pull request merged**